mindsdb_native/libs/phases/model_analyzer/model_analyzer.py

```diff
@@ -90,9 +94,9 @@ def run(self):
 
         for col in output_columns:
             pval = ProbabilisticValidator(col_stats=self.transaction.lmd['stats_v2'][col], col_name=col, input_columns=input_columns)
-            predictions_arr = [normal_predictions] + [empty_input_predictions[col] for col in ignorable_input_columns]
+            predictions_arr = [normal_predictions_test] + [empty_input_predictions_test[col] for col in ignorable_input_columns]
```

Comment:
1. aren't these square brackets in the first term making a list whose first item is another list?
2. making a new list and calling `list.extend() `instead of adding lists might be a lot faster for many predictions.

Reply:
yeap, this is indeed, it's an array of arrays of predictions.